src/main/scala/com/snowflake/snowpark/internal/OpenTelemetry.scala

```diff
@@ -60,51 +60,33 @@ object OpenTelemetry extends Logging {
       execName: String,
       execHandler: String,
       execFilePath: String)(func: => T): T = {
-    try {
-      spanInfo.withValue[T](spanInfo.value match {
-        // empty info means this is the entry of the recursion
-        case None =>
```

Comment:
In the old code, the nest actions will emit same span multiple times. This fix skips the emit process in the nested action.